unumpy/_multimethods.py

```diff
@@ -784,6 +824,69 @@ def broadcast_to(array, shape, subok=False):
     return (array,)
 
 
+def _normalize_axis(ndim, axis, allow_repeats=False):
+    if isinstance(axis, int):
+        axis = (axis,)
+    if isinstance(axis, (tuple, list)):
+        axis_pos = [ndim + ax if ax < 0 else ax for ax in axis]
+
+    if (len(axis) != len(set(axis_pos))) and not allow_repeats:
```

Comment:
I would put the quick check first, that way short circuiting avoids the need to create the `set` object:
```suggestion
    if not allow_repeats and (len(axis) != len(set(axis_pos))):
```